### PR TITLE
18204 disable restore based on ff

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#18204 

Added feature flag

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-583-qrxozjdm.web.app

**Comment by PaulGarewal** [comment]
Looks good from what I understand. 

Are the tests supposed to fail? I will wait for confirmation/update on the tests before approval.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:139]: Did you confirm this with UI/UX Team? (Yui or Janis)

File: src/mixins/allowable-actions-mixin.ts
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:136]: "test"?

File: src/mixins/allowable-actions-mixin.ts
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:141]: FYI, did you know that if you check the ff first, and it's false, then the expression will short-circuit and return right away (thus not checking the other conditions)?

**Review by severinbeauvais — APPROVED** [review]
OK with me as long as UI/UX team is OK with displaying the menu option conditionally (which is different from now).

**Comment by severinbeauvais** [comment]
Please fix the broken tests. They're probably related to you making the dropdown menu item conditional.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [reply src/components/Dashboard/StaffNotation.vue:139]: I heard that Yui is OK with this change. Resolving.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-583-qrxozjdm.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-583-qrxozjdm.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-583-qrxozjdm.web.app

**Comment by severinbeauvais** [comment]
> OK with me as long as UI/UX team is OK with displaying the menu option conditionally (which is different from now).

I've heard that they are OK with this.

Note that all 3 restoration filings are now hidden when the business is not historical.

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:263]: If I understand correctly, what this is doing is providing a mock for GetFeatureFlag() but we don't care about the response, because the unit tests below just verify that the menu items are displayed, not whether they are enabled.

**Review by severinbeauvais — COMMENTED** [review]
Looks great but there's still 1 unit test error.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-583-qrxozjdm.web.app

**Review by JazzarKarim — APPROVED** [review]
Great job James 👍

**Pull request merged**
